sdks/unity/AgonesSdk.cs

```diff
@@ -73,6 +73,31 @@ private void OnApplicationQuit()
         #endregion
 
         #region AgonesRestClient Public Methods
+
+        /// <summary>
+        /// Async method that waits to connect to the SDK Server. Will timeout
+        /// and return false after 30 seconds.
+        /// </summary>
+        /// <returns>A task that indicated whether it was successful or not</returns>
+        public async Task<bool> Connect()
+        {
+            for (var i = 0; i < 30; i++)
+            {
+                Log($"Attempting to connect...{i + 1}");
+                var gameServer = await GameServer();

```

Comment:
try-catch for when GameServer throws exception to retry on the exception as well?

Reply:
Ooh good idea. I assume I can catch the generic [Exception](https://docs.microsoft.com/en-us/dotnet/api/system.exception?view=netframework-4.8) to catch everything?

Reply:
yes.

Reply:
Done!